pallets/gear/src/internal.rs

```diff
@@ -585,18 +617,50 @@ where
                 })
                 .unwrap_or_default();
 
+            let delay_gas_value = delay
+                .saturating_add(CostsPerBlockOf::<T>::reserve_for().unique_saturated_into())
+                .saturating_mul(CostsPerBlockOf::<T>::dispatch_stash().unique_saturated_into());
+
+            let mut gas_amount = delay_gas_value as u64;
+
             // Message is going to be inserted into mailbox.
             //
             // No hold bound checks required, because gas_limit isn't less than threshold.
             if gas_limit >= threshold {
                 // Cutting gas for storing in mailbox.
-                GasHandlerOf::<T>::cut(sender_node, dispatch.id(), gas_limit)
+                gas_amount += gas_limit;
+                GasHandlerOf::<T>::cut(sender_node, dispatch.id(), gas_amount)
                     .unwrap_or_else(|e| unreachable!("GasTree corrupted! {:?}", e));
 
                 // TODO: adapt this line if gasful sending appears for reservations (#1828)
                 if let Some(reservation_id) = reservation {
                     Self::remove_gas_reservation_with_task(dispatch.source(), reservation_id);
                 }
+                is_mailboxed = true;
```

Comment:
why not just separate `gas_limit >= threshold` condition into var above ?

Reply:
done